src/editor/particle_settings_widget.cpp

```diff
@@ -0,0 +1,150 @@
+//  SuperTux
+//  Copyright (C) 2020 A. Semphris <semphris@protonmail.com>
+//
+//  This program is free software: you can redistribute it and/or modify
+//  it under the terms of the GNU General Public License as published by
+//  the Free Software Foundation, either version 3 of the License, or
+//  (at your option) any later version.
+//
+//  This program is distributed in the hope that it will be useful,
+//  but WITHOUT ANY WARRANTY; without even the implied warranty of
+//  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+//  GNU General Public License for more details.
+//
+//  You should have received a copy of the GNU General Public License
+//  along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+#if 0
+#include "editor/particle_settings_widget.hpp"
+
+#include <math.h>
+
+#include "editor/editor.hpp"
+#include "video/drawing_context.hpp"
+#include "video/renderer.hpp"
+#include "video/video_system.hpp"
+#include "video/viewport.hpp"
+
+ParticleSettingsWidget::ParticleSettingsWidget() :
+  m_scrolling(),
+  m_hovering(),
+  m_total_region(),
+  m_covered_region(),
+  m_progress(),
+  m_rect(),
+  m_scaled_rect(),
+  is_horizontal(),
+  last_mouse_pos(),
+  zoom_factor()
+{
+  m_covered_region = VideoSystem::current()->get_viewport().get_rect().get_height();
+  m_total_region = 2000;
```

Comment:
You can initialize `m_total_region` in the initializer list.

Reply:
`particle_settings_widget` is a half-complete cyborg of random thoughts and recycled code, which resulted in a never-developed melting pot of random content.

In brief, it wasn't kept in the final product :D I removed the .cpp and the .hpp from my next commit.